File: languages/go/entity/integrity/record_receipt.go
package integrity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// RecordReceipt represents a receipt for a record, including anchor ID, client, record, and status information.
type RecordReceipt struct {
	Anchor int64
	Client string
	Record string
	Status string
}

func NewRecordReceiptFromProto(r *proto.RecordReceipt) RecordReceipt {
	if r == nil {
		return RecordReceipt{}
	}
	return RecordReceipt{
		Anchor: r.Anchor,
		Client: r.Client,
		Record: r.Record,
		Status: r.Status,
	}
}

File: languages/js/src/entity/integrity/record_receipt.ts
import * as proto from "../../bridge/proto/integrity_entities";

/**
 * Represents a receipt for a record, including anchor ID, client, record, and status information.
 */
export class RecordReceipt {
  anchor: number;
  client: string;
  record: string;
  status: string;

  /**
   * Constructs a RecordReceipt object with the specified parameters.
   * @param anchor 
   * @param client 
   * @param record 
   * @param status 
   */
  constructor(anchor: number, client: string, record: string, status: string) {
    this.anchor = anchor;
    this.client = client;
    this.record = record;
    this.status = status;
  }

  static fromProto(r: proto.RecordReceipt): RecordReceipt {
    return new RecordReceipt(r.anchor, r.client, r.record, r.status);
  }

  toProto(): proto.RecordReceipt {
    return proto.RecordReceipt.fromPartial({
      anchor: this.anchor,
      client: this.client,
      record: this.record,
      status: this.status
    });
  }
}
